.github/workflows/minor_version_release.yml

```diff
@@ -25,17 +21,47 @@ jobs:
           python -m pip install --upgrade pip
           pip install pydoc-markdown==4.5.1
 
-      - name: Convert version name to regular semver (e.g. v1.10.x -> v1.10.0)
-        id: regular_semver
-        env:
-          version: ${{github.event.inputs.version}}
+      - name: Get current version
+        id: current-version
+        shell: bash
+        # We only need `major.minor` in Readme so we cut the full version string to the first two tokens
+        run: echo "::set-output name=minor::$(cat VERSION.txt | cut -d "." -f 1,2)"
+
+      - name: Create new version branch
+        run: |
+          git checkout -b v${{ steps.current-version.outputs.minor }}.x
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git push -u origin v${{ steps.current-version.outputs.minor }}.x
+
+      - name: Bump version on main
+        id: bump-version
+        run: |
+          git fetch
+          git checkout main
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git pull
+          python ./.github/utils/bump_version.py
+          echo "::set-output name=new_minor::$(cat VERSION.txt | cut -d "." -f 1,2)"
+          git checkout -b bump-version
+          git add ./VERSION.txt
+          git commit -m "Bump version"
+          git push -u origin bump-version
```

Comment:
we should push to `main` here, correct?

Reply:
After our discussion, I was under the impression that we didn't want a workflow to push directly to main and that we were aiming to open a PR that can be merged into main. This code creates that branch, but doesn't open the PR. The PR creation will need to be done manually

Reply:
Right! my bad...
You can create a PR with an additional step like this:
```
    - name: Create Pull Request
      run: gh pr create -B main -H bump-version --title 'Bump unstable version' --body 'Created by Github action'
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```